frontend/src/metabase-types/api/collection.ts

```diff
@@ -48,7 +48,8 @@ type CollectionItemModel =
   | "dashboard"
   | "pulse"
   | "snippet"
-  | "collection";
+  | "collection"
+  | "indexed-entity";
```

Comment:
Adding `indexed-entity` here since we're checking for `indexed-entity` in the `SearchResults` component. In the future, we might want to think about the relationship between the `CollectionItemModel` and `SearchModelType`